source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -598,6 +604,23 @@ Otherwise, the `Connection <#connection>`_ is marked as available.
     else:
       mark connection as available
 
+Determining if a Connection is "perished"
+-----------------------------------------
+
+Before checking a `Connection`_ in or out, the pool MUST
+ensure the `Connection`_ is not "perished". A `Connection`_ is
+considered "perished" if it is at least one of the following:
+
+-  **Stale:** The `Connection`_ 's generation does not match the generation of the parent pool
+-  **Idle:** The `Connection`_ is currently "available" and has been for longer than **maxIdleTimeMS**.
+-  **Errored:** The `Connection`_ has experienced an error that indicates it is no longer recommended for use. To determine if a `Connection`_ is in such a state, the pool MUST check if it is “live” by using poll(), select(), or similar functionality available in the language’s networking library. This check MUST NOT block. This can be achieved by passing a timeout of 0 to poll(), for example. Examples of how a `Connection`_ might enter such a state include, but are not limited to: 
```

Comment:
This section seems to mix errors that happen when the connection is being actively used with errors that happen while the connection is idle in the pool. On a first read, this makes it seem like we should call poll() even after a client side error occurs while using the connection (eg a socket timeout) . It might be clearer to distinguish these two cases. Should add a new term here like **Dead**?:
```
...
-  **Idle:** The `Connection`_ is currently "available" and has been for longer than **maxIdleTimeMS**.
-  **Errored:** The `Connection`_ has experienced an error that indicates it is no longer recommended for use.  Examples include, but are not limited to: ...
-  **Dead:** The `Connection`_ is currently "available" (i.e. idle in the pool) and has experienced an error. To determine if a `Connection`_ is in such a state, the pool MUST check if it is “live” by using poll(), select(), or similar functionality available in the language’s networking ...
```

Reply:
Yeah, I like that distinction, updated to that. One question now is do we need another reason for a connection to be closed? Or should we still repurpose the "error" reason for closing? I think error still fits here, but having a separate "dead" close reason also seems applicable.